src/aro/Parser.zig

```diff
@@ -323,7 +323,7 @@ fn expectIdentifier(p: *Parser) Error!TokenIndex {
         return p.errExpectedToken(.identifier, actual);
     }
 
-    return (try p.eatIdentifier()) orelse unreachable;
+    return (try p.eatIdentifier()) orelse error.ParsingFailed;
```

Comment:
Removing the `catch unreachables` makes sense but the reason this can't use `.?` is because `eatIdentifier` returns null when an extended identifier contains errors. Shouldn't it return `error.ParsingFailed` instead or just the token with errors? Seems like returning null leads to redundant diagnostics in many places.